src/components/sectionList/translation/TranslationForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { InputFieldFF } from '@dhis2/ui'
+import React, { useState } from 'react'
+import { Field, Form, useFormState } from 'react-final-form'
+import {
+    BaseListModel,
+    getTranslatedProperty,
+    useSchemaFromHandle,
+} from '../../../lib'
+import { camelCaseToConstantCase } from '../../../lib/utils'
+import { constantCaseToCamelCase } from '../../../lib/utils/caseTransformers'
+import { Query, WrapQueryResponse } from '../../../types'
+import { WebLocale, Translation } from '../../../types/generated'
+import { LoadingSpinner } from '../../loading/LoadingSpinner'
+
+type DBLocalesResponse = WrapQueryResponse<WebLocale[]>
+
+export const useDBLocales = () => {
+    const query = useDataQuery<DBLocalesResponse>({
+        result: {
+            resource: 'locales/db',
+        },
+    })
+
+    return {
+        ...query,
+        data: query.data?.result,
+    }
+}
+
+const useLocales = (modelId: string) => {
+    const schema = useSchemaFromHandle()
+    const [query] = useState<Query>(() => ({
+        result: {
+            resource: `${schema.plural}`,
+            id: `${modelId}/translations`,
+        },
+    }))
+
+    const queryResult =
+        useDataQuery<WrapQueryResponse<{ translations: Translation[] }>>(query)
+
+    return {
+        ...queryResult,
+        data: queryResult.data?.result,
+    }
+}
+
+const useInitialValues = (model: BaseListModel, selectedLocale: WebLocale) => {
```

Comment:
Might want to rename to `useInitialValuesQuery`, and return the query as well. Then wrap the from with `Loader` component to handle error states.